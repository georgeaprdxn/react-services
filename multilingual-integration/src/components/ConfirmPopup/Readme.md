ConfirmPopup

```jsx inside Markdown
import Overlay from './example/confirmPopup.example.tsx'
;<Overlay />
```
