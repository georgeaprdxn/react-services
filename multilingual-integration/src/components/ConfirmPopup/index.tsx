import React from 'react'

import {
	ConfirmBoxWrapper,
	OkButton,
	ConfirmBox,
	QuestionText,
	ButtonWrapper,
	CloseIcon,
	CloseIconWrapper,
	CancelButton,
	ConfirmBoxOverlay,
} from './styled'

type OverlayProps = {
	/**
	 * handler for negative button
	 */
	onNegativeHandler: () => void
	/**
	 * handler for positive button
	 */
	onPositiveHandler: (
		e: React.MouseEvent<HTMLButtonElement>,
		okButtonRef: React.RefObject<HTMLButtonElement>
	) => void
	/**
	 * positive button text
	 */
	positiveText: string
	/**
	 * negative button text
	 */
	negativeText: string
	/**
	 * handler for close button
	 */
	onPopupClose: () => void
	/**
	 * warning text to be displayed on popup
	 */
	warningText: string
} & typeof defaultProps

const Overlay = (props: OverlayProps) => {
	const okButtonRef = React.useRef<HTMLButtonElement>(null)

	const {
		positiveText,
		negativeText,
		warningText,
		onNegativeHandler,
		onPositiveHandler,
	} = props

	const onPopupClose = () => {
		return props.onPopupClose && props.onPopupClose()
	}

	return (
		<ConfirmBoxWrapper>
			<ConfirmBoxOverlay
				onClick={onPopupClose}
				className="custom-popup-close"
			/>
			<ConfirmBox>
				<QuestionText>{warningText}</QuestionText>
				<CloseIconWrapper>
					<CloseIcon onClick={onPopupClose} />
				</CloseIconWrapper>
				<ButtonWrapper>
					{positiveText && (
						<OkButton
							ref={okButtonRef}
							onClick={(e: React.MouseEvent<HTMLButtonElement>) =>
								onPositiveHandler(e, okButtonRef)
							}
						>
							{positiveText}
						</OkButton>
					)}
					{negativeText && (
						<CancelButton onClick={onNegativeHandler}>
							{negativeText}
						</CancelButton>
					)}
				</ButtonWrapper>
			</ConfirmBox>
		</ConfirmBoxWrapper>
	)
}

const defaultProps = {
	negativeText: 'Cancel',
	positiveText: 'Ok',
	warningText: 'Do you want to continue ?',
}

Overlay.defaultProps = defaultProps

export default Overlay
