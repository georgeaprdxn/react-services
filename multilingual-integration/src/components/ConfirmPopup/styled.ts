import { Close } from '../../assets/icons'
import styled, { css } from 'styled-components'

const SharedDeleteCss = css`
	width: 20%;
	opacity: 0.8;
	border-radius: 3px;
	border: solid 1px rgba(216, 216, 216, 0.56);
	display: inline-block;
	font-size: 12px;
	font-weight: bold;
	line-height: 1.42;
	color: #363636;
	padding: 25px 10px 23px;
	cursor: pointer;
	&:hover {
		background-color: #8b8b8b;
		transition: all 0.7s;
		color: #fff;
		opacity: 1;
	}
`

const ConfirmBoxWrapper = styled.div``
const OkButton = styled.button`
	${SharedDeleteCss}
	margin-top: 46px;
	margin-bottom: 43px;
	padding: 10px;
	background-color: #fff;
	@media (max-width: 968px) {
		margin-top: 45px;
	}
`

const ConfirmBox = styled.div`
	box-sizing: border-box;
	width: 70%;
	max-width: 769px;
	padding: 0 20px;
	text-align: center;
	position: fixed;
	margin: 0 auto;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
	background-color: #fff;
	z-index: 20;
	@media (max-width: 968px) {
		width: 518px;
	}
`

const QuestionText = styled.p`
	font-size: 22px;
	font-weight: bold;
	padding-top: 42px;
	color: #363636;
	opacity: 0.7;
	@media (max-width: 968px) {
		font-size: 25px;
	}
`

const ButtonWrapper = styled.div`
	width: 67%;
	margin: 0 auto;
`

const CloseIcon = styled(Close)`
	transform: translate(-50%, -50%);
	margin-left: 50%;
	margin-top: 50%;
	width: 26px;
	height: 26px;
`

const CloseIconWrapper = styled.div`
	position: absolute;
	top: -14px;
	right: -13px;
	background-color: #fff;
	width: 34px;
	border-radius: 50%;
	height: 34px;
	cursor: pointer;
`

const CancelButton = styled.button`
	${SharedDeleteCss}
	margin: 46px 0 43px 28px;
	padding: 10px;
	background-color: #fff;
	@media (max-width: 968px) {
		margin-top: 45px;
	}
`

const ConfirmBoxOverlay = styled.div`
	position: fixed;
	top: 0;
	bottom: 0;
	left: 0;
	right: 0;
	background-color: #979797;
	z-index: 20;
`
export {
	ConfirmBoxWrapper,
	OkButton,
	ConfirmBox,
	QuestionText,
	ButtonWrapper,
	CloseIcon,
	CloseIconWrapper,
	CancelButton,
	ConfirmBoxOverlay,
}
