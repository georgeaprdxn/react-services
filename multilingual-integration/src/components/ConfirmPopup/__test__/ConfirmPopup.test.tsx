import React from 'react'
import { shallow } from 'enzyme'

import Overlay from '../example/confirmPopup.example'
import mockConfirmPopupData from './ConfirmPopup.mock.json'
import ConfirmPopup from '../index'

describe('ConfirmPopup component tests', () => {
	let ConfirmPopupComp: any = null

	beforeEach(() => {
		ConfirmPopupComp = shallow(<Overlay />)
	})

	it('should render a button', () => {
		expect(ConfirmPopupComp.find('.custom-button').length).toEqual(1)
	})

	it('calls open popup', () => {
		ConfirmPopupComp.find('.custom-button').simulate('click')
		expect(ConfirmPopupComp.state('showPopup')).toBe(true)
	})

	it('validate child component props', () => {
		ConfirmPopupComp.find('.custom-button').simulate('click')

		expect(ConfirmPopupComp.find(ConfirmPopup).props().warningText).toEqual(
			mockConfirmPopupData.warningText
		)
		expect(ConfirmPopupComp.find(ConfirmPopup).props().positiveText).toEqual(
			mockConfirmPopupData.positiveText
		)
		expect(ConfirmPopupComp.find(ConfirmPopup).props().negativeText).toEqual(
			mockConfirmPopupData.negativeText
		)
	})

	it('calls close popup', () => {
		ConfirmPopupComp.instance().onPopupClose()
		expect(ConfirmPopupComp.state('showPopup')).toBe(false)
	})

	it('matches the snaphot', () => {
		expect(ConfirmPopupComp).toMatchSnapshot()
	})
})
