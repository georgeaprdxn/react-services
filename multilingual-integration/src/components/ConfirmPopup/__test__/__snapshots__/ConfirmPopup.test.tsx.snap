// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPopup component tests matches the snaphot 1`] = `
<Fragment>
  <button
    className="custom-button"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    Show Confirm popup
  </button>
</Fragment>
`;
